### PR TITLE
Implement Token Saving Enhancement and Logout Command

### PR DESCRIPTION
**Pull request by eshanvaid**
## PR Description

This pull request aims to resolve GaloyMoney/galoy-cli#121. The following changes have been made to ensure the flow works as expected:

- [x] Remove `GALOY_JWT` and its verification .
- [x] Save the login token to `~/.galoy-cli/GALOY_TOKEN` after fetching it, overriding any existing token stored in the `.galoy-cli` folder.
- [x] Retrieve the token from the `GALOY_TOKEN` file instead of environment variable.
- [x] Create a `galoy-cli logout` command to facilitate the removal of the login token. This command deletes the `GALOY_TOKEN` file.

### CODE REVIEW

**Comment by nicolasburtey** [comment]
GALOY_JWT should be renamed to GALOY_TOKEN because we no longer use a JWT

File: src/main.rs [hunk 1]
```diff
@@ -26,7 +33,7 @@ struct Cli {
     #[clap(short, long, value_parser, default_value_t = false)]
     debug: bool,
 
-    #[clap(short, long, value_parser, env = "GALOY_JWT", hide_env_values = true)]
+    #[clap(short, long, value_parser, default_value = "None")]
     jwt: Option<String>,
```
Comment (sandipndev) [inline src/main.rs:37]: lets rename all `jwt`s to token.

Comment (eshanvaid) [reply src/main.rs]: Addressed in 4c2d705d7fa44243e00efc4db61d265866c716a0

File: src/main.rs [hunk 2]
```diff
@@ -114,8 +127,19 @@ fn main() -> anyhow::Result<()> {
...
                 .user_login(phone, code)
                 .context("issue logging in")?;
+            token::save_token(&token_file, &result)?;
             println!("{:#?}", result);
         }
+        Commands::Logout => {
+            if token_file.exists() {
+                fs::remove_file(&token_file).with_context(|| {
+                    format!("failed to delete token file '{}'", token_file.display())
+                })?;
```
Comment (sandipndev) [inline src/main.rs:137]: This is a token based operation so it should be scoped to `token.rs`

File: src/token.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        )
+    })?;
+
+    let mut file = File::create(token_file)
+        .with_context(|| format!("failed to create file '{}'", token_file.display()))?;
+
+    file.write_all(token.as_bytes())
+        .with_context(|| format!("failed to write to file '{}'", token_file.display()))?;
+
+    println!("Token saved to {}", token_file.display());
```
Comment (sandipndev) [inline src/token.rs:27]:
```suggestion
    info!("Token saved to {}", token_file.display());
```

**Review by sandipndev — APPROVED** [review]

**Pull request merged**